<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>500</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <family>Arial</family>
   </font>
  </property>
  <property name="cursor">
   <cursorShape>PointingHandCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>Triangulation_demo_3</string>
  </property>
  <property name="windowIcon">
   <iconset resource="T3_demo.qrc">
    <normaloff>:/T3_demo/icons/icons/cgal_logo.xpm</normaloff>:/T3_demo/icons/icons/cgal_logo.xpm</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="Viewer" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1100</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoad_Points"/>
    <addaction name="actionSave_Points"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionGenerate_Points"/>
    <addaction name="separator"/>
    <addaction name="actionIncremental_Construct"/>
    <addaction name="actionStop_Animation"/>
    <addaction name="separator"/>
    <addaction name="actionClear_Scene"/>
   </widget>
   <widget class="QMenu" name="menuMode">
    <property name="title">
     <string>Mode</string>
    </property>
    <addaction name="actionNormal_View"/>
    <addaction name="actionInsert_Vertex"/>
    <addaction name="actionInsert_Point"/>
    <addaction name="actionSelect_Vertex"/>
    <addaction name="actionMove_Vertex"/>
    <addaction name="actionFind_NearestNb"/>
    <addaction name="actionEmpty_Sphere"/>
   </widget>
   <widget class="QMenu" name="menuShow">
    <property name="title">
     <string>Show</string>
    </property>
    <addaction name="actionShow_Axis"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Vertex"/>
    <addaction name="actionShow_DEdge"/>
    <addaction name="actionShow_VEdge"/>
    <addaction name="actionShow_Facet"/>
    <addaction name="separator"/>
    <addaction name="actionFlat"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionDemo_Help"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuMode"/>
   <addaction name="menuShow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Points"/>
   <addaction name="actionSave_Points"/>
   <addaction name="separator"/>
   <addaction name="actionGenerate_Points"/>
   <addaction name="actionClear_Scene"/>
   <addaction name="separator"/>
   <addaction name="actionIncremental_Construct"/>
   <addaction name="actionStop_Animation"/>
   <addaction name="separator"/>
   <addaction name="actionShow_Axis"/>
   <addaction name="separator"/>
   <addaction name="actionFlat"/>
   <addaction name="separator"/>
   <addaction name="actionShow_Vertex"/>
   <addaction name="actionShow_DEdge"/>
   <addaction name="actionShow_VEdge"/>
   <addaction name="actionShow_Facet"/>
   <addaction name="separator"/>
   <addaction name="actionNormal_View"/>
   <addaction name="actionInsert_Vertex"/>
   <addaction name="actionInsert_Point"/>
   <addaction name="actionSelect_Vertex"/>
   <addaction name="actionMove_Vertex"/>
   <addaction name="actionFind_NearestNb"/>
   <addaction name="actionEmpty_Sphere"/>
   <addaction name="separator"/>
   <addaction name="actionPreferences"/>
   <addaction name="separator"/>
   <addaction name="actionQuit"/>
  </widget>
  <action name="actionGenerate_Points">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/pointRandom.png</normaloff>:/T3_demo/icons/icons/pointRandom.png</iconset>
   </property>
   <property name="text">
    <string>Generate Points</string>
   </property>
   <property name="toolTip">
    <string>Generate Points</string>
   </property>
   <property name="statusTip">
    <string>Generate Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionLoad_Points">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/fileOpen.png</normaloff>:/T3_demo/icons/icons/fileOpen.png</iconset>
   </property>
   <property name="text">
    <string>Load Points...</string>
   </property>
   <property name="toolTip">
    <string>Load Points from a file</string>
   </property>
   <property name="statusTip">
    <string>Load Points from a file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_Points">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/fileSave.png</normaloff>:/T3_demo/icons/icons/fileSave.png</iconset>
   </property>
   <property name="text">
    <string>Save Points...</string>
   </property>
   <property name="toolTip">
    <string>Save points to a file</string>
   </property>
   <property name="statusTip">
    <string>Save points to a file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionShow_Axis">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/coordinates.jpeg</normaloff>:/T3_demo/icons/icons/coordinates.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Axis</string>
   </property>
   <property name="toolTip">
    <string>Show/Hide Axis</string>
   </property>
   <property name="statusTip">
    <string>Show/Hide Axis</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/quit.jpeg</normaloff>:/T3_demo/icons/icons/quit.jpeg</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionClear_Scene">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/clear.jpeg</normaloff>:/T3_demo/icons/icons/clear.jpeg</iconset>
   </property>
   <property name="text">
    <string>Clear Scene</string>
   </property>
   <property name="toolTip">
    <string>Clear Scene</string>
   </property>
   <property name="statusTip">
    <string>Clear Scene</string>
   </property>
  </action>
  <action name="actionShow_Vertex">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/show_point.jpeg</normaloff>:/T3_demo/icons/icons/show_point.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Vertices</string>
   </property>
   <property name="toolTip">
    <string>Show Vertices</string>
   </property>
   <property name="statusTip">
    <string>Show Vertices</string>
   </property>
  </action>
  <action name="actionShow_DEdge">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/show_delaunay.jpeg</normaloff>:/T3_demo/icons/icons/show_delaunay.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Delaunay Edges</string>
   </property>
   <property name="toolTip">
    <string>Show Delaunay edges</string>
   </property>
   <property name="statusTip">
    <string>Show Delaunay edges</string>
   </property>
  </action>
  <action name="actionShow_VEdge">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/show_voronoi.jpeg</normaloff>:/T3_demo/icons/icons/show_voronoi.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Voronoi Edges</string>
   </property>
   <property name="toolTip">
    <string>Show Voronoi edges</string>
   </property>
   <property name="statusTip">
    <string>Show Voronoi edges</string>
   </property>
  </action>
  <action name="actionShow_Facet">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/show_facet.jpeg</normaloff>:/T3_demo/icons/icons/show_facet.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Facets</string>
   </property>
   <property name="toolTip">
    <string>Show Delaunay Facets</string>
   </property>
   <property name="statusTip">
    <string>Show Delaunay Facets</string>
   </property>
  </action>
  <action name="actionFlat">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/flat.png</normaloff>
     <normalon>:/T3_demo/icons/icons/stereo.png</normalon>:/T3_demo/icons/icons/flat.png</iconset>
   </property>
   <property name="text">
    <string>Flat</string>
   </property>
   <property name="toolTip">
    <string>Toggle 3D effect</string>
   </property>
   <property name="statusTip">
    <string>Toggle 3D effect</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/preferences.jpeg</normaloff>:/T3_demo/icons/icons/preferences.jpeg</iconset>
   </property>
   <property name="text">
    <string>Preferences...</string>
   </property>
   <property name="toolTip">
    <string>Change Colors, Transparency, etc.</string>
   </property>
   <property name="statusTip">
    <string>Change Colors, Transparency, etc.</string>
   </property>
  </action>
  <action name="actionInsert_Vertex">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/insert.jpeg</normaloff>:/T3_demo/icons/icons/insert.jpeg</iconset>
   </property>
   <property name="text">
    <string>Insert Vertex</string>
   </property>
   <property name="toolTip">
    <string>Insert a point and update the triangulation</string>
   </property>
   <property name="statusTip">
    <string>Insert a point and update the triangulation</string>
   </property>
  </action>
  <action name="actionInsert_Point">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/insert_point.jpg</normaloff>:/T3_demo/icons/icons/insert_point.jpg</iconset>
   </property>
   <property name="text">
    <string>Insert Point</string>
   </property>
   <property name="toolTip">
    <string>Show the conflict region of a point and insert it</string>
   </property>
   <property name="statusTip">
    <string>Show the conflict region of a point and insert it</string>
   </property>
  </action>
  <action name="actionSelect_Vertex">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/select_hand.jpeg</normaloff>:/T3_demo/icons/icons/select_hand.jpeg</iconset>
   </property>
   <property name="text">
    <string>Select Vertex</string>
   </property>
   <property name="toolTip">
    <string>Select vertices</string>
   </property>
   <property name="statusTip">
    <string>Select vertices</string>
   </property>
  </action>
  <action name="actionMove_Vertex">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/move_1.jpeg</normaloff>:/T3_demo/icons/icons/move_1.jpeg</iconset>
   </property>
   <property name="text">
    <string>Move Vertex</string>
   </property>
   <property name="toolTip">
    <string>Move a vertex</string>
   </property>
   <property name="statusTip">
    <string>Move a vertex</string>
   </property>
  </action>
  <action name="actionFind_NearestNb">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/nearest_nb.png</normaloff>:/T3_demo/icons/icons/nearest_nb.png</iconset>
   </property>
   <property name="text">
    <string>Nearest Neighbor Search</string>
   </property>
   <property name="toolTip">
    <string>Find the nearest neighbor of the query point</string>
   </property>
   <property name="statusTip">
    <string>Find the nearest neighbor of the query point</string>
   </property>
  </action>
  <action name="actionEmpty_Sphere">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/empty_sphere.jpeg</normaloff>:/T3_demo/icons/icons/empty_sphere.jpeg</iconset>
   </property>
   <property name="text">
    <string>Show Empty Sphere</string>
   </property>
   <property name="iconText">
    <string>Click to select a cell and show empty sphere of that cell.</string>
   </property>
   <property name="toolTip">
    <string>Locate the query point in a cell and show empty sphere of that cell</string>
   </property>
   <property name="statusTip">
    <string>Locate the query point in a cell and show empty sphere of that cell</string>
   </property>
  </action>
  <action name="actionNormal_View">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/normal_view.jpeg</normaloff>:/T3_demo/icons/icons/normal_view.jpeg</iconset>
   </property>
   <property name="text">
    <string>Normal Mode</string>
   </property>
   <property name="toolTip">
    <string>Normal Mode</string>
   </property>
   <property name="statusTip">
    <string>Normal Mode</string>
   </property>
  </action>
  <action name="actionDemo_Help">
   <property name="text">
    <string>Triangulation_3D Demo Help</string>
   </property>
   <property name="toolTip">
    <string>Triangulation_3D Demo Help</string>
   </property>
   <property name="statusTip">
    <string>Triangulation_3D Demo Help</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionIncremental_Construct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/play.jpeg</normaloff>
     <normalon>:/T3_demo/icons/icons/pause.jpeg</normalon>:/T3_demo/icons/icons/play.jpeg</iconset>
   </property>
   <property name="text">
    <string>Insertion Animation</string>
   </property>
   <property name="toolTip">
    <string>Animation of incremental Delaunay triangulation</string>
   </property>
   <property name="statusTip">
    <string>Animation of incremental Delaunay triangulation</string>
   </property>
  </action>
  <action name="actionStop_Animation">
   <property name="icon">
    <iconset resource="T3_demo.qrc">
     <normaloff>:/T3_demo/icons/icons/stop.jpeg</normaloff>:/T3_demo/icons/icons/stop.jpeg</iconset>
   </property>
   <property name="text">
    <string>Stop Animation</string>
   </property>
   <property name="toolTip">
    <string>Stop Animation</string>
   </property>
   <property name="statusTip">
    <string>Stop Animation</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="3" margin="1"/>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header>Viewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="T3_demo.qrc"/>
 </resources>
 <connections/>
</ui>
